apollo-router/src/query_planner/dual_query_planner.rs

```diff
@@ -208,16 +213,33 @@ fn subscription_primary_matches(this: &SubscriptionNode, other: &SubscriptionNod
 }
 
 fn operation_matches(this: &SubgraphOperation, other: &SubgraphOperation) -> bool {
-    operation_without_whitespace(this) == operation_without_whitespace(other)
-}
-
-fn operation_without_whitespace(op: &SubgraphOperation) -> String {
-    op.as_serialized().replace([' ', '\n'], "")
+    let this_ast = match ast::Document::parse(this.as_serialized(), "this_operation.graphql") {
```

Comment:
flagging that reparsing an operation can potentially be expensive, but since this is run in a background thread, we are likely okay.

Reply:
JS plan has to be parsed, since it comes as a string, but the rust plan could be directly converted from `executable::Document`. I'm considering to do that.